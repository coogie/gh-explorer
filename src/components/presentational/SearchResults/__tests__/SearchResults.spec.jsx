import React from "react";

import { SearchResults } from "../";

const mockProps = {
  isLoading: false,
  results: [
    {
      name: "react",
      html_url: "https://github.com/facebook/react",
      description:
        "A declarative, efficient, and flexible JavaScript library for building user interfaces.",
      updated_at: "2018-12-05T22:27:34Z",
      homepage: "https://reactjs.org",
      stargazers_count: 117028,
      stargazers_count_formatted: "117,028",
      owner: {
        login: "facebook",
        avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4",
        html_url: "https://github.com/facebook"
      }
    },
    {
      name: "react-native",
      html_url: "https://github.com/facebook/react-native",
      description: "A framework for building native apps with React.",
      updated_at: "2018-12-05T23:05:39Z",
      homepage: "https://facebook.github.io/react-native/",
      stargazers_count: 71617,
      stargazers_count_formatted: "71,617",
      owner: {
        login: "facebook",
        avatar_url: "https://avatars3.githubusercontent.com/u/69631?v=4",
        html_url: "https://github.com/facebook"
      }
    }
  ]
};

describe("SearchResults Component", () => {
  it("should render a list of results", () => {
    const wrapper = shallow(<SearchResults {...mockProps} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should display a count of how many results are visible", () => {
    const wrapper = shallow(<SearchResults {...mockProps} />);
    const list = wrapper.find(".SearchResults__list");

    expect(list.prop("header")).toBe("Showing 2 results");
  });

  it("should NOT display a count of how many results are visible when no results", () => {
    const props = {
      ...mockProps,
      results: []
    };
    const wrapper = shallow(<SearchResults {...props} />);
    const list = wrapper.find(".SearchResults__list");

    expect(list.prop("header")).toBe(null);
  });
});
