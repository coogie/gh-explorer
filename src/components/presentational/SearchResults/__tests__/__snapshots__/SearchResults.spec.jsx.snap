// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults Component should render a list of results 1`] = `"<div class=\\"ant-list SearchResults__list ant-list-vertical ant-list-split\\"><div class=\\"ant-list-header\\">Showing 2 results</div><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-list-item SearchResults__item\\"><div class=\\"ant-list-item-meta SearchResults__meta\\"><div class=\\"ant-list-item-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"https://avatars3.githubusercontent.com/u/69631?v=4\\"/></span></div><div class=\\"ant-list-item-meta-content\\"><h4 class=\\"ant-list-item-meta-title\\"><a href=\\"https://github.com/facebook/react\\" rel=\\"noopener\\">react <small>(facebook)</small></a></h4></div></div><div class=\\"ant-list-item-content\\">A declarative, efficient, and flexible JavaScript library for building user interfaces.</div><ul class=\\"ant-list-item-action\\"><li><i style=\\"margin-right:8px\\" class=\\"anticon anticon-star-o\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"star\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z\\"></path></svg></i>117,028</li></ul></div><div class=\\"ant-list-item SearchResults__item\\"><div class=\\"ant-list-item-meta SearchResults__meta\\"><div class=\\"ant-list-item-meta-avatar\\"><span class=\\"ant-avatar ant-avatar-circle ant-avatar-image\\"><img src=\\"https://avatars3.githubusercontent.com/u/69631?v=4\\"/></span></div><div class=\\"ant-list-item-meta-content\\"><h4 class=\\"ant-list-item-meta-title\\"><a href=\\"https://github.com/facebook/react-native\\" rel=\\"noopener\\">react-native <small>(facebook)</small></a></h4></div></div><div class=\\"ant-list-item-content\\">A framework for building native apps with React.</div><ul class=\\"ant-list-item-action\\"><li><i style=\\"margin-right:8px\\" class=\\"anticon anticon-star-o\\"><svg viewBox=\\"64 64 896 896\\" class=\\"\\" data-icon=\\"star\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z\\"></path></svg></i>71,617</li></ul></div></div></div></div>"`;
